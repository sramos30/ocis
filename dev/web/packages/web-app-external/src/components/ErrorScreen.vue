<template>
  <div class="oc-text-center">
    <oc-icon size="xxlarge" name="error-warning" fill-type="line" />
    <p v-if="message" class="oc-text-lead">{{ message }}</p>
  </div>
</template>

<script>
export default {
  name: 'ErrorScreen',
  props: {
    /**
     * Error message passed from external app
     */
    message: {
      default: '',
      type: String,
      required: false
    }
  }
}
</script>
