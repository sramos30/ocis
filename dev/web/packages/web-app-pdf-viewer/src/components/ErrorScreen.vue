<template>
  <div class="oc-position-center oc-text-center">
    <oc-icon size="xxlarge" name="error-warning" fill-type="line" />
  </div>
</template>

<script>
export default {
  name: 'ErrorScreen'
}
</script>
