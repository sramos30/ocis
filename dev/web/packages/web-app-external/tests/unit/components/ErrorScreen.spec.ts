import ErrorScreen from '../../../src/components/ErrorScreen.vue'
import { Wrapper, mount, createLocalVue } from '@vue/test-utils'
import GetTextPlugin from 'vue-gettext'

const localVue = createLocalVue()

localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

let wrapper: Wrapper<any>

describe('The external app error screen component', () => {
  test('displays an icon and a paragraph', () => {
    wrapper = mount(ErrorScreen, {
      localVue,
      stubs: {
        OcIcon: true
      },
      propsData: {
        message: 'Error when loading the application'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
