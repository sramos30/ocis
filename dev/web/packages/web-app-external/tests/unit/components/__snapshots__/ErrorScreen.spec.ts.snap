// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The external app error screen component displays an icon and a paragraph 1`] = `
<div class="oc-text-center">
  <ocicon-stub size="xxlarge" name="error-warning" fill-type="line"></ocicon-stub>
  <p class="oc-text-lead">Error when loading the application</p>
</div>
`;
